### PR TITLE
Add support for fine-tunning and files using the Azure API.

### PR DESCRIPTION
**Pull request by sorinsuciu-msft**

### CODE REVIEW

**Review by kennyhsu5 — APPROVED** [review]

**Comment by ChrisHMSFT** [comment]
@kennyhsu5  saw this was approved. One quick question -- we don't have merge rights. are you able to merge or do we need another approval? thanks!

File: openai/api_resources/abstract/api_resource.py
Comment (sorinsuciu-msft) [reply openai/api_resources/abstract/api_resource.py:36]: Someone changed in on of the previous PRs the api_prefix to be "" which in turn causes the result to have "//" hence my change. I don't know if the previous change was meant to be permanent or temporary but for now, it only works like this.

File: README.md
Comment (sorinsuciu-msft) [reply README.md:80]: Txs for spotting it.

File: openai/api_resources/abstract/api_resource.py
Comment (kennyhsu5) [reply openai/api_resources/abstract/api_resource.py:36]: Not sure I understand this completely, do you have an example of what call breaks/causes the "//" issue?

I did a quick search of the repo, and I don't see anything setting api_prefix. AFAIK, your changes also only touches azure_api_prefix, so not sure why this case would be reached to begin with. Seems like it should skip this case and return url of `"/%ss" % (base)` instead.

If it does reach this case, did api_prefix and OBJECT_NAME have an extra "/"? Otherwise kind of confused what would cause it to have "//"

**Comment by kennyhsu5** [comment]
> @kennyhsu5 saw this was approved. One quick question -- we don't have merge rights. are you able to merge or do we need another approval? thanks!

I think I can merge it. Just trying to get more clarity on one final comment/issue first.

File: openai/api_resources/abstract/api_resource.py
Comment (sorinsuciu-msft) [reply openai/api_resources/abstract/api_resource.py:36]: api_prefix used to be "v1", a constant defined in this file above. It was changed in the following commit to "": https://github.com/sorinsuciu-msft/openai-python/commit/b39bddd99dc243bf09bec60f4866e2fb542d0e8f. When it was the constant "v1" it makes sense for the return to be: "/%s/%ss" % (cls.api_prefix, base) as it will return /v1/fine-tunes as the relative path for example. Now that it is just the empty string, it just returns "//fine-tunes" and hence the double /.

Comment (sorinsuciu-msft) [reply openai/api_resources/abstract/api_resource.py]: It seems to work as expected now. As you said I might have had a strange OBJECT_NAME during my testing. Reverting.

**Comment by kennyhsu5** [comment]
@sorinsuciu-msft Can you double check the unit tests are passing with this PR?

I'm seeing test_file_upload and test_engine_search_url_composition_azure_no_operation failures when trying to incorporate this PR into our CI.

Reverting it [here](https://github.com/openai/openai-python/pull/79) since CI seem to be passing fine without this change.

**Pull request merged**
